traitlets/traitlets.py

```diff
@@ -489,11 +488,28 @@ def _validate(self, obj, value):
         return value
 
     def _cross_validate(self, obj, value):
-        if hasattr(obj, '_%s_validate' % self.name):
-            cross_validate = getattr(obj, '_%s_validate' % self.name)
-            value = cross_validate(value, self)
+        try:
+            cb = getattr(obj, '_%s_validate' % self.name)
+        except AttributeError:
+            cb = obj._trait_validators.get(self.name,None)
+        else:
+            warn("_[traitname]_validate handlers are deprecated: use register_validator instead",
+                 DeprecationWarning, stacklevel=2)
+            if not callable(cb):
```

Comment:
Thanks for pointing out the consistency issues.  Personally, I would eliminate the extra code checking for callable.  It makes the code simpler, and essentially the same error will be raised by python automatically if it isn't callable.